engine/rpcserver.go

```diff
@@ -1213,3 +1221,292 @@ func (s *RPCServer) GetAuditEvent(ctx context.Context, r *gctrpc.GetAuditEventRe
 
 	return &resp, nil
 }
+
+// GCTScriptStatus returns a slice of current running scripts that includes next run time and uuid
+func (s *RPCServer) GCTScriptStatus(ctx context.Context, r *gctrpc.GCTScriptStatusRequest) (*gctrpc.GCTScriptStatusResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptStatusResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	totalVM := len(gctscript.AllVMs)
+
+	if totalVM < 1 {
+		return &gctrpc.GCTScriptStatusResponse{Status: "no scripts running"}, nil
+	}
+
+	resp := &gctrpc.GCTScriptStatusResponse{
+		Status: fmt.Sprintf("%v of %v virtual machines running", totalVM, gctscript.GCTScriptConfig.MaxVirtualMachines),
+	}
+
+	for x := range gctscript.AllVMs {
+		resp.Scripts = append(resp.Scripts, &gctrpc.GCTScript{
+			UUID:    gctscript.AllVMs[x].ID.String(),
+			Name:    gctscript.AllVMs[x].ShortName(),
+			NextRun: gctscript.AllVMs[x].NextRun.String(),
+		})
+	}
+
+	return resp, nil
+}
+
+func (s *RPCServer) GCTScriptQuery(ctx context.Context, r *gctrpc.GCTScriptQueryRequest) (*gctrpc.GCTScriptQueryResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptQueryResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	resp := &gctrpc.GCTScriptQueryResponse{}
+
+	UUID, err := uuid.FromString(r.Script.UUID)
+	if err != nil {
+		return &gctrpc.GCTScriptQueryResponse{Status: "error"}, nil
+	}
+
+	if v, f := gctscript.AllVMs[UUID]; f {
```

Comment:
the UUID being valid or not has nothing to do with the information this is a basic map lookup changing to a false condition check would just mean if its not found in the map...

It returns empty because of the empty struct init at @1258 this was by design no script running with that uuid = don't return any data

Reply:
I was more interested in a simple "UUID not found" if someone puts in UUID which doesn't exist, similar to what you've done here https://github.com/thrasher-corp/gocryptotrader/pull/383/files#diff-9a26a62ed888d995d75ee011e28aa4e6R1332